OsmAnd-java/src/main/java/net/osmand/router/GeneralRouter.java

```diff
@@ -731,6 +731,12 @@ public double calculateTurnTime(RouteSegment segment, RouteSegment prev) {
 				totalPenalty += getLeftTurn();
 			} else if (diff > Math.PI / 3) {
 				totalPenalty += getRightTurn();
+			} else if (diff > Math.PI / 3 / 2) {
+				// increase sensitivity for highway *_link
```

Comment:
totalPenalty += getRightTurn() / 2;

Reply:
refactored to linear penalty for >30 degree